lightning-background-processor/src/lib.rs

```diff
@@ -632,7 +663,8 @@ mod tests {
 		let data_dir = nodes[0].persister.get_data_dir();
 		let persister = Arc::new(Persister::new(data_dir));
 		let event_handler = |_: &_| {};
-		let bg_processor = BackgroundProcessor::start(persister, event_handler, nodes[0].chain_monitor.clone(), nodes[0].node.clone(), nodes[0].net_graph_msg_handler.clone(), nodes[0].peer_manager.clone(), nodes[0].logger.clone());
+		let scorer = Arc::new(Mutex::new(test_utils::TestScorer::with_penalty(0)));
```

Comment:
Here and later tests, if the test doesn't explicitly check scoring, can we pass in `None` instead? Alternatively, add it the test `Node` struct so it only needs to be created in one place.

Reply:
Done. I wasn't able to use just `None` because the compiler wasn't able to infer the type:

> cannot infer type for type parameter `S` declared on the associated function

so I added `None::<Arc<Mutex<FixedPenaltyScorer>>>`. 

I know `FixedPenaltyScorer` is deprecated, however, when I tried `None::<Arc<Mutex<ProbabilisticScorer<NetworkGraph, TestLogger>>>`, the compiler complains:

```
669 | ...e(), None::<Arc<Mutex<ProbabilisticScorer<NetworkGraph, Logger>>>>);
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Deref` is not implemented for `NetworkGraph`
```

is that a problem?

also, should we create a follow-up issue for updating `TestScorer` to return a `ProbabilisticScorer` instead of `FixedPenaltyScorer`?

Reply:
> Done. I wasn't able to use just `None` because the compiler wasn't able to infer the type:
> 
> > cannot infer type for type parameter `S` declared on the associated function
> 
> so I added `None::<Arc<Mutex<FixedPenaltyScorer>>>`.
> 

Hmm... right, that's a bit verbose. I'd say add a `scorer` field to `Node`  and do `nodes[0].scorer.clone()`.

> I know `FixedPenaltyScorer` is deprecated, however, when I tried `None::<Arc<Mutex<ProbabilisticScorer<NetworkGraph, TestLogger>>>`, the compiler complains:
> 
> ```
> 669 | ...e(), None::<Arc<Mutex<ProbabilisticScorer<NetworkGraph, Logger>>>>);
>     |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Deref` is not implemented for `NetworkGraph`
> ```
> 
> is that a problem?
> 

With this approach you would need to use `&NetworkGraph` or `Arc<NetworkGraph>` since the type parameter must implement `Deref`.

> also, should we create a follow-up issue for updating `TestScorer` to return a `ProbabilisticScorer` instead of `FixedPenaltyScorer`?

No need. `Scorer` is deprecated not, `FixedPenaltyScorer`.

Reply:
@jkczyz gotcha. I guess if `FixedPenaltyScorer` is not deprecated I won't change it to `ProbabilisticScorer` then. Just added the scorer field to Node. should be ready for review again :) thanks!